.github/workflows/ci.yml

```diff
@@ -0,0 +1,60 @@
+################################################################################
+#  Licensed to the Apache Software Foundation (ASF) under one
+#  or more contributor license agreements.  See the NOTICE file
+#  distributed with this work for additional information
+#  regarding copyright ownership.  The ASF licenses this file
+#  to you under the Apache License, Version 2.0 (the
+#  "License"); you may not use this file except in compliance
+#  with the License.  You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+# limitations under the License.
+################################################################################
+
+name: Build flink-connector-elasticsearch
+on: [push, pull_request]
+jobs:
+  compile_and_test:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        jdk: [8, 11]
+    env:
+      MVN_GOALS: clean install
```

Comment:
By using `clean` in the test step you end up re-doing everything from the compile step.

I think there are only 2 approaches that really make sense. 
1) `mvn test-compile -T<parallelism> -> mvn install`
2) `mvn clean install`

1) `test-compile` only compiles the code but doesn't create jars. This is useful because it can be run in parallel, and jars are inevitable recreated in the install step anyway. To me this only makes sense if you have a good reason for this separation though. In Flink we do this because we have several test jobs to skip as much as possible from the compile phase; I don't see any gain for this project, but I also don't know what your plans are.


On another note, consider only running `mvn verify` so you don't need to worry about snapshot artifacts ending up being cached in the maven repository.

Reply:
> To me this only makes sense if you have a good reason for this separation though. In Flink we do this because we have several test jobs to skip as much as possible from the compile phase; I don't see any gain for this project, but I also don't know what your plans are.

I think it's important to fail fast; if the compilation can't succeed because of a checkstyle violation or a dependency convergence, I think you want to have that checked asap before starting to run tests. 

I did do a couple of test runs with either running two steps (first a `mvn clean test-compile -T2` followed by `mvn install`) or in one go (running `mvn clean install`). 

However, running two steps appears to make the pipeline slower: almost 9 minutes vs 7 minutes. 

Example runs:
* Two steps: [run 1](https://github.com/MartijnVisser/flink-connector-elasticsearch/actions/runs/2095531366/attempts/4), [run 2](https://github.com/MartijnVisser/flink-connector-elasticsearch/actions/runs/2095531366/attempts/3), [run 3](https://github.com/MartijnVisser/flink-connector-elasticsearch/actions/runs/2095531366/attempts/2)
* One step: [run 1](https://github.com/MartijnVisser/flink-connector-elasticsearch/actions/runs/2095579462/attempts/3), [run 2](https://github.com/MartijnVisser/flink-connector-elasticsearch/actions/runs/2095579462/attempts/2), [run 3](https://github.com/MartijnVisser/flink-connector-elasticsearch/actions/runs/2095579462/attempts/1)

I'll update it to one step for now. We can always reconsider for the future